universal-application-tool-0.0.1/test/services/applicant/question/ApplicantQuestionTest.java

```diff
@@ -123,63 +28,62 @@ public void errorsPresenterExtendedForAllTypes(QuestionType type)
-        new ApplicantQuestion(numberQuestionDefinition, new ApplicantData());
+        new ApplicantQuestion(
+            QuestionDefinitionBuilder.sample(QuestionType.NUMBER).build(), new ApplicantData());
     assertThat(numberApplicantQuestion.createNumberQuestion()).isInstanceOf(NumberQuestion.class);
 
+    ApplicantQuestion radioApplicantQuestion =
+        new ApplicantQuestion(
+            QuestionDefinitionBuilder.sample(QuestionType.RADIO_BUTTON).build(),
+            new ApplicantData());
+    assertThat(radioApplicantQuestion.createSingleSelectQuestion())
+        .isInstanceOf(SingleSelectQuestion.class);
+
     ApplicantQuestion singleSelectApplicantQuestion =
-        new ApplicantQuestion(dropdownQuestionDefinition, new ApplicantData());
+        new ApplicantQuestion(
+            QuestionDefinitionBuilder.sample(QuestionType.DROPDOWN).build(), new ApplicantData());
     assertThat(singleSelectApplicantQuestion.createSingleSelectQuestion())
         .isInstanceOf(SingleSelectQuestion.class);
 
     ApplicantQuestion textApplicantQuestion =
-        new ApplicantQuestion(textQuestionDefinition, new ApplicantData());
+        new ApplicantQuestion(
+            QuestionDefinitionBuilder.sample(QuestionType.TEXT).build(), new ApplicantData());
     assertThat(textApplicantQuestion.createTextQuestion()).isInstanceOf(TextQuestion.class);
-
-    ApplicantQuestion radioApplicantQuestion =
-        new ApplicantQuestion(radioButtonQuestionDefinition, new ApplicantData());
-    assertThat(radioApplicantQuestion.createSingleSelectQuestion())
-        .isInstanceOf(SingleSelectQuestion.class);
-
-    ApplicantQuestion checkboxApplicantQuestion =
-        new ApplicantQuestion(checkboxQuestionDefinition, new ApplicantData());
-    assertThat(checkboxApplicantQuestion.createMultiSelectQuestion())
-        .isInstanceOf(MultiSelectQuestion.class);
   }
 
   @Test
-  public void equals() {
+  public void equals() throws UnsupportedQuestionTypeException {
+    AddressQuestionDefinition addressQuestionDefinition =
+        (AddressQuestionDefinition) QuestionDefinitionBuilder.sample(QuestionType.ADDRESS).build();
+    TextQuestionDefinition textQuestionDefinition =
+        (TextQuestionDefinition) QuestionDefinitionBuilder.sample(QuestionType.TEXT).build();
     ApplicantData dataWithAnswers = new ApplicantData();
     dataWithAnswers.putString(Path.create("applicant.color"), "blue");
 
     new EqualsTester()
         .addEqualityGroup(
             new ApplicantQuestion(addressQuestionDefinition, new ApplicantData()),
             new ApplicantQuestion(addressQuestionDefinition, new ApplicantData()))
-        .addEqualityGroup(
-            new ApplicantQuestion(dropdownQuestionDefinition, new ApplicantData()),
-            new ApplicantQuestion(dropdownQuestionDefinition, new ApplicantData()))
-        .addEqualityGroup(
-            new ApplicantQuestion(nameQuestionDefinition, new ApplicantData()),
```

Comment:
(blocking) why are these equality checks being deleted?

Reply:
I can add them back. I had deleted them because it felt like it might get excessive to add an equality test for every single question type ever. I thought it was more relevant to test at least two different question types against each other as well as a question with and without applicant data.

LMK if you think I should add them back (and add all the missing question types that weren't here before).

Reply:
To simplify it, can this look something more like 

```
EqualsTester equalsTester = new EqualsTester()
ImmutabeList<QuestionDefinition> questionDefinitions = ImmutableList.of(...); // all the question definitions here.
for(QuestionDefinition questionDefinition : questionDefinitions) {
  equalsTester.addEqualityGroup(...)
}
equalsTester.testEquals();
```

Reply:
I think we should have full coverage here. I think it is confusing to have some coverage, but not all, especially if the cost is pretty low.

Reply:
That SGTM. Will update on Monday.

Reply:
Done